src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -224,22 +229,53 @@ public DashboardWebApplication(Action<WebApplicationBuilder>? configureBuilder =
         _app.MapGrpcService<OtlpLogsService>();
     }
 
+    private static void HandleConfigurationError(ILogger<DashboardWebApplication> logger, string[] failureMessages)
+    {
+        WriteVersion(logger);
+        logger.LogError("Failed to start the dashboard due to {Count} configuration error(s).", failureMessages.Length);
+        foreach (var message in failureMessages)
+        {
+            logger.LogError("{ErrorMessage}", message);
+        }
+
+        Environment.Exit(-1);
```

Comment:
I don't think we want to do this from the host though. It's a library, not an executable.

Reply:
I don't understand. Would you prefer to set the exit code in `Program.cs` instead of `DashboardWebApplication.cs`?

Reply:
Or are you saying that if we have similar code for printing out config errors in the host then it shouldn't Environment.Exit?

Reply:
> Or are you saying that if we add similar code for printing out config errors in the host then it shouldn't Environment.Exit?

This, but we need to unwind.

Since the dashboard owns main, this should happen in main.

Reply:
I agree. Having it where it currently is caused test failures anyway.